### ISSUE TITLE
ill-timed Rails ActionCable requests consume

### ISSUE THREAD

**Issue by agraves** [wardencommunity/warden#182]

**Comment by agraves** [comment]
So, doing a little more research on this. Looks like my `app/channels/application_cable/connection.rb` was like this:

```
+    def find_verified_user
+      if env['warden'].user
+        env['warden'].user
+      else
+        reject_unauthorized_connection
+      end
+    end
```

And doing a bit more research, it looks like the recommendation is to do something like this:

```
      def find_verified_user
        if verified_user = User.find_by(id: cookies.encrypted[:user_id])
          verified_user
        else
          reject_unauthorized_connection
        end
      end
```
I guess my question now is -- would any invocation of `env['warden'].user` consume whatever state `login_as` puts it in? i.e. is this a "just don't screw with env['warden'].user` directly" issue? Or is there something about the `ActionCable` request (like the asset requests?) that predisposes it to eat these cookies?

**Comment by jsmestad** [comment]
@agraves yeah it sounds difficult to track down. How would you suppose we approach options 2 or 3 without adding additional dependencies from Rails?

If you have any ideas, we'd be open to a PR.

**Comment by MatheusRich** [comment]
This is likely related to https://github.com/wardencommunity/warden/issues/163, so not only action table but also turbo streams could cause this.

**Comment by jsmestad** [comment]
@MatheusRich feel free to stage a PR

**Comment by MatheusRich** [comment]
@jsmestad I'm trying to work on a solution, although I don't know exactly what to do.

**Comment by MatheusRich** [comment]
@jsmestad I've done this https://github.com/MatheusRich/warden/commit/4dd89fa5040ca60a1ffcfd88dc355697f31cd2b4. It looks like it fixed it for me in my apps. I have very little context with warden, so please tell me if that makes sense. If so, I'll open a PR here.
